### DISCUSSION TITLE
Cross-platform mechanism for setting the mouse cursor

### DISCUSSION THREAD

**Discussion by mikeparker104** [CommunityToolkit/Maui#938] [New Feature Discussions]
Provide the ability to change the mouse cursor when hovering over a View on desktop platforms.

Such an API would provide a way to set the cursor type for a given View using a cross-platform enum of cursor types available on both macOS and Windows. A key considerations is the selection of cursor types is limited on MacCatalyst compared to AppKit. However, it's technically possible to set the unsupported/undocumented AppKit cursor types using selectors e.g., [Cocoa predefined resize mouse cursor?](https://stackoverflow.com/questions/27242353/cocoa-predefined-resize-mouse-cursor).

The cross-platform cursor type enum might include the following values:

```csharp
public enum CursorType
{
    Arrow,
    Hand,
    Help,
    IBeam,
    NotAllowed,
    Wait,
    Crosshair,
    UpDown,
    LeftRight,
    LeftUpRightDown,
    LeftDownRightUp
}
```

Each cross-platform **CursorType** value would need mapping to the respective platform-specific cursor that is the best fit. For example:

| CursorType            | Mac                                                              | Windows                         |
| -------------------- | ---------------------------------------------- | --------------------------- |
| Arrow                    | arrow                                                           | Arrow                                |
| Hand                     | pointingHand                                              | Hand                                 |
| Help                      | _helpCursor                                                 | Help                                  |
| IBeam                    | IBeamCursor                                               | IBeam                                |
| NotAllowed           | operationNotAllowed                                 | UniversalNo                      |
| Wait                       | _waitCursor                                                 | Wait                                  |
| Cross                     | crosshair                                                      | Cross                                |
| UpDown                | resizeUpDown                                             | SizeNorthSouth                |
| LeftRight               | resizeLeftRight                                             | SizeWestEast                    |
| LeftUpRightDown | _windowResizeNorthWestSouthEastCursor | SizeNorthwestSoutheast |
| LeftDownRightUp | _windowResizeNorthEastSouthWestCursor | SizeNortheastSouthwest |

It would be great to explore what this API could look like as part of this toolkit and whether the cursor could be set via an attached property and/or behaviour. Any input is appreciated.

**Comment by VladislavAntonyuk** [comment]
https://vladislavantonyuk.github.io/articles/Setting-a-cursor-for-.NET-MAUI-VisualElement

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
